File: old/actions/install_haproxy.py
"""
Install HAProxy action
"""
import logging
from cli import Apt
from .base import Action
logger = logging.getLogger(__name__)

class InstallHaproxyAction(Action):
    """Action to install HAProxy package"""
    description = "haproxy installation"

    def execute(self) -> bool:
        """Install HAProxy package"""
        if not self.apt_service:
            logger.error("APT service not initialized")
            return False
        logger.info("Installing haproxy package...")
        install_cmd = Apt().install(["haproxy"])
        output = self.apt_service.execute(install_cmd)
        if output is None:
            logger.error("haproxy installation failed")
            return False
        return True

File: src/actions/install_haproxy.go
package actions

import (
	"enva/libs"
	"enva/services"
)

// InstallHaproxyAction installs HAProxy package
type InstallHaproxyAction struct {
	*BaseAction
}

func NewInstallHaproxyAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &InstallHaproxyAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *InstallHaproxyAction) Description() string {
	return "haproxy installation"
}

func (a *InstallHaproxyAction) Execute() bool {
	if a.APTService == nil {
		libs.GetLogger("install_haproxy").Printf("APT service not initialized")
		return false
	}
	libs.GetLogger("install_haproxy").Printf("Installing haproxy package...")
	output, exitCode := a.APTService.Install([]string{"haproxy"})
	if exitCode == nil || *exitCode != 0 {
		libs.GetLogger("install_haproxy").Printf("haproxy installation failed: %s", output)
		return false
	}
	return true
}

